pagefaultgames/pokerogue
# [Bug][Beta] Double vouchers fix

## What are the changes the user will see?
Hopefully, users will not get double golden vouchers.  

## Why am I making these changes?
This bug is blocking beta. 

## What are the changes from a developer perspective?
PR #1940 :  This fixed the issue of 1st Classic Clear voucher ids from being undefined. However, recent changes from #3618 led to this problem resurfacing, meaning that user accounts be considered to have two 1st Classic Clear vouchers - one with an actual id, the other with an undefined id. When title phase validated the vouchers, it would validate the newly created 'undefined' 1st Classic Clear id, which held the same information as the one as the actual id, and give the account another voucher.

This was fixed by moving initAchievements() above initVouchers() since initVouchers() depends on initAchievements(). Then, a conditional was added to the 1st classic clear achievement to prevent future validations of the achievement / associated voucher. This doesn't interfere with 1st classic clears. However, although this prevents vouchers from generating an 'undefined' 1st classic clear voucher again, those who were present during the accidental beta merge from last week may already have a valid 'undefined' one and still receive a second golden voucher.  

## How to test the changes?
Check the contents of vouchers -> console.log()
Check if a new user is able to clear classic and receive the achievement/voucher. 

## Checklist
